### PR TITLE
Replace clangll.rs with clang-sys dependency

### PR DESCRIPTION
**Pull request by KyleMayes**
Hello,

This pull request replaces the file `clangll.rs` with my [clang-sys](http://kylemayes.com/clang-sys/3_4/clang_sys/) crate.

Support for clang 3.4.* is retained, and my bindings have better type safety (enums are actually enums, the flags are actual bitflags, etc.).

I don't know if this change is welcome, it's alright if it isn't. It only took me about 15-20 minutes to make this change. :)

---
This change is [Reviewable](https://reviewable.io/reviews/crabtw/rust-bindgen/301)

### CODE REVIEW

**Comment by Yamakaky** [comment]
https://github.com/crabtw/rust-bindgen/issues/227 ;)

File: Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,7 @@ keywords = [ "bindings", "ffi", "code-generation" ]
 build = "build.rs"
 
 [dependencies]
+clang-sys = { version = "0.3.*", features = ["clang_3_4"] }
```
Comment (Yamakaky) [inline Cargo.toml]: Why 3.4? Is it a minimum supported version?

Comment (KyleMayes) [inline Cargo.toml]: 3.4.x is the lowest version supported by `clang-sys`, but now that I think about it, requesting it as a feature doesn't actually do anything. I'll change this line.

**Comment by Yamakaky** [comment]
TGTM. @nox?

**Comment by Yamakaky** [comment]
@KyleMayes Do you think it would be easy/useful to also replace clang.rs by clang-rs?

**Comment by nox** [comment]
We can't merge this as such for two reasons:

* What happens to LIBCLANG_PATH?
* Does clang_sys actually try to find libclang by itself at all?

**Comment by Yamakaky** [comment]
It currently dyn links https://github.com/KyleMayes/clang-sys/blob/master/src/lib.rs#L1139. What about integrating https://github.com/crabtw/rust-bindgen/blob/master/build.rs in you crate @KyleMayes?

**Comment by KyleMayes** [comment]
I'll do that.

**Comment by KyleMayes** [comment]
OK, I replicated the functionality of `build.rs` in `clang-sys`. The `static` feature is handed off to `clang-sys`.

**Comment by nox** [comment]
You should also probably steal our Travis support.

**Comment by Yamakaky** [comment]
Looks like the issue raised by @nox is solved. Except dependency version update, anything else before merging ?

**Comment by nox** [comment]
@KyleMayes Would you be willing to add us as collaborators on clang-sys?

**Comment by nox** [comment]
And how is clang-sys written? By hand? Is to just use bindgen to generate it a long term goal?

**Comment by Yamakaky** [comment]
Inception ^^

**Comment by Yamakaky** [comment]
Could you update the clang_sys version?

**Comment by Yamakaky** [comment]
Merged, thank you!

**Pull request closed without merging**